sefaria/helper/topic.py

```diff
@@ -107,6 +107,61 @@ def group_links_by_type(link_class, links, annotate_links, group_related):
                 grouped_links[link_type_slug]['pluralTitle'] = link_type.get('pluralDisplayName', is_inverse)
     return grouped_links
 
+def merge_props_for_similar_refs(curr_link, new_link):
+    # when grouping similar refs, make sure the source to display has the maximum curatedPrimacy of all the similar refs,
+    # as well as datasource and descriptions of all the similar refs
+    data_source = new_link.get('dataSource', None)
+    if data_source:
+        update_refs(curr_link, new_link, data_source)
+        update_data_source_in_link(curr_link, new_link, data_source)
```

Comment:
curr_link is being changed and not returned which is a little difficult to follow, please change it so that the dictionary is returned and the next function uses the new dictionary